picard/ui/options/renaming.py

```diff
@@ -117,35 +118,49 @@ def __init__(self, parent=None):
                 self.toggle_file_renaming
             )
         )
-        self.ui.file_naming_format.textChanged.connect(self.check_formats)
-        self.ui.file_naming_format_default.clicked.connect(self.set_file_naming_format_default)
+        self.ui.open_script_editor.clicked.connect(self.show_script_editing_page)
         self.ui.move_files_to_browse.clicked.connect(self.move_files_to_browse)
 
-        script_edit = self.ui.file_naming_format
+        self.ui.example_filename_after.itemSelectionChanged.connect(self.match_before_to_after)
+        self.ui.example_filename_before.itemSelectionChanged.connect(self.match_after_to_before)
+
+        script_edit = self.ui.move_additional_files_pattern
         self.script_palette_normal = script_edit.palette()
         self.script_palette_readonly = QPalette(self.script_palette_normal)
         disabled_color = self.script_palette_normal.color(QPalette.Inactive, QPalette.Window)
         self.script_palette_readonly.setColor(QPalette.Disabled, QPalette.Base, disabled_color)
-        self.ui.scripting_documentation_button.clicked.connect(self.show_scripting_documentation)
-        self.ui.example_filename_sample_files_button.clicked.connect(self._sample_example_files)
-        self._sampled_example_files = []
-
-        # Sync example lists vertical scrolling
-        def sync_vertical_scrollbars(widgets):
-            """Sync position of vertical scrollbars for listed widgets"""
-            def _sync_scrollbar_vert(widget, value):
-                widget.blockSignals(True)
-                widget.verticalScrollBar().setValue(value)
-                widget.blockSignals(False)
-
-            widgets = set(widgets)
-            for widget in widgets:
-                for other in widgets - {widget}:
-                    widget.verticalScrollBar().valueChanged.connect(
-                        partial(_sync_scrollbar_vert, other))
-
-        # Sync example lists vertical scrolling
-        sync_vertical_scrollbars((self.ui.example_filename_before, self.ui.example_filename_after))
+
+        self.ui.example_filename_sample_files_button.clicked.connect(self.update_example_files)
+
+        self.examples = ScriptEditorExamples(self, self)
```

Comment:
Why `(self, self)` here?

```python
def __init__(self, parent=None, tagger=None):
```

https://github.com/metabrainz/picard/pull/1792/files#diff-b826bd41b92baa93cfac5658a55109f4926829b68a630d63a2ac342c17c0980fR91

Reply:
Actually, the `parent` argument was never used so I removed it.  Used keywords as you suggested and now pass the `tagger` object directly rather than the parent of the `tagger` option.